src/compiler/checker.ts

```diff
@@ -18901,7 +18901,7 @@ namespace ts {
                         }
                     }
 
-                    const isPerformingCommonPropertyChecks = (relation !== comparableRelation || relation === comparableRelation && isLiteralType(source)) &&
+                    const isPerformingCommonPropertyChecks = (relation !== comparableRelation || !(source.flags & TypeFlags.Union) && isLiteralType(source)) &&

```

Comment:
I think this is equivalent

```suggestion
                    const isPerformingCommonPropertyChecks = (relation !== comparableRelation || isUnitType(source)) &&
```

Reply:
Are enums unit types?

Reply:
They are

https://github.com/microsoft/TypeScript/pull/51140#discussion_r993857211